engine/api/workflow/dao_run.go

```diff
@@ -313,6 +314,40 @@ func loadRun(db gorp.SqlExecutor, withArtifacts bool, query string, args ...inte
 	return &wr, nil
 }
 
+// canBeRun return boolean to know if a wokrflow node run can be run or not
```

Comment:
should we refactor the workflow.Process function to call this ?

Reply:
Ok, I suggest testing for a moment in production (it's less critical) and if it's stable it could be put in process.go to make some refactor :) 
I've added a TODO

Reply:
I'm fine with it